src/libraries/GetTradeableOrder.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.24;
+
+import {Math} from "lib/openzeppelin/contracts/utils/math/Math.sol";
+import {IConditionalOrder, GPv2Order, IERC20} from "lib/composable-cow/src/BaseConditionalOrder.sol";
+
+library GetTradeableOrder {
+    /// @dev Avoid stack too deep errors with `getTradeableOrder`.
+    struct GetTradeableOrderParams {
+        address pool;
+        IERC20 token0;
+        IERC20 token1;
+        uint256 priceNumerator;
+        uint256 priceDenominator;
+        bytes32 appData;
+    }
+
+    /// @notice The largest possible duration of any AMM order, starting from the current block timestamp.
+    uint32 public constant MAX_ORDER_DURATION = 5 * 60;
+
+    function getTradeableOrder(GetTradeableOrderParams memory params)
```

Comment:
Assuming the math in this function was copied from the existing code.

Reply:
Absolutely, with the exclusion of changing out one of the sub commands which was revert guarded to make it play well previously with the watch-tower.